base/compiler/tfuncs.jl

```diff
@@ -2638,13 +2638,15 @@ function return_type_tfunc(interp::AbstractInterpreter, argtypes::Vector{Any}, s
 
     if length(argtypes) == 3
         aft = widenslotwrapper(argtypes[2])
-        if !isa(aft, Const) && !(isType(aft) && !has_free_typevars(aft)) &&
-                !(isconcretetype(aft) && !(aft <: Builtin))
-            return UNKNOWN
-        end
         argtypes_vec = Any[aft, af_argtype.parameters...]
     else
         argtypes_vec = Any[af_argtype.parameters...]
+        isempty(argtypes_vec) && push!(argtypes_vec, Union{})
+        aft = argtypes_vec[1]
+    end
+    if !(isa(aft, Const) || (isType(aft) && !has_free_typevars(aft)) ||
+            (isconcretetype(aft) && !(Builtin <: aft) && !iskindtype(aft)))
```

Comment:
```suggestion
            (isconcretetype(aft) && !(aft <: Builtin) && !iskindtype(aft)))
```
shouldn't this condition flipped?

Reply:
no, the concrete type goes on the left

Reply:
ah, wait, yeah, `aft` is the concrete type